File: src/main/java/kitchenpos/menus/dto/MenuCreateRequest.java
package kitchenpos.menus.dto;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public record MenuCreateRequest(
        String name,
        BigDecimal price,
        UUID menuGroupId,
        boolean displayed,
        List<MenuProductCreateRequest> menuProducts
) {
}

File: src/main/java/kitchenpos/menus/tobe/dto/in/MenuCreateRequest.kt
package kitchenpos.menus.tobe.dto.`in`

import kitchenpos.common.Price
import java.util.UUID

data class MenuCreateRequest(
    val name: String,
    val price: Price,
    val menuGroupId: UUID,
    val displayed: Boolean,
    val menuProducts: List<MenuProductCreateRequest>
) {
    val productIds: List<UUID>
        get() = menuProducts.map { it.productId }
}

data class MenuProductCreateRequest(
    val productId: UUID,
    val quantity: Long
)
